services/spar/src/Spar/Scim/User.hs

```diff
@@ -65,27 +65,54 @@ import qualified Web.Scim.Schema.User             as Scim.User (schemas)
 -- UserDB instance
 
 instance Scim.UserDB SparTag Spar where
-  -- | List all users, possibly filtered by some predicate.
+  -- | Filters users based on some predicate.  We do not support this endpoint
+  -- without an explicit predicate due to performance reasons.  The predicate
+  -- can currently look for users based on handle or SSO ID. 
+  --
+  -- This endpoint will look for users in Brig, and then synthesize the
+  -- corresponding SCIM record.
+  -- look for SSO users in brig; and then return the corresponding SCIM record.
+  -- getUsers itself will not do any mutations in the SCIM database; as it
+  -- feels weird to me (arian) that a GET causes side-effects like that.
+  -- Instead, 'updateUser' will when updating a user see if the user was
+  -- already in the SCIM Table; and if not insert it on demand, just like
+  -- 'POST'.
   getUsers :: ScimTokenInfo
            -> Maybe Scim.Filter
            -> Scim.ScimHandler Spar (Scim.ListResponse (Scim.StoredUser SparTag))
-  getUsers ScimTokenInfo{stiTeam} mbFilter = do
-    members <- lift $ getTeamMembers stiTeam
-    brigusers :: [User]
-      <- filter (not . userDeleted) <$>
-         lift (Intra.Brig.getBrigUsers ((^. Galley.userId) <$> members))
-    scimusers :: [Scim.StoredUser SparTag]
-      <- lift . wrapMonadClient . Data.getScimUsers $ BrigTypes.userId <$> brigusers
-    let check user = case mbFilter of
-          Nothing -> pure True
-          Just filter_ ->
-            let user' = Scim.value (Scim.thing user)
-            in case Scim.filterUser filter_ user' of
-                 Right res -> pure res
-                 Left err  -> throwError $
-                   Scim.badRequest Scim.InvalidFilter (Just err)
-    -- FUTUREWORK: once bigger teams arrive, we should have pagination here.
-    Scim.fromList <$> filterM check scimusers
+  getUsers _ Nothing = do
+    -- TODO(arianvp): This shouldn't be a 501 according to joe
```

Comment:
yes.  the rule: "if it should wake up admins at night even if it happens a lot, don't make it 500."

Reply:
We deviate from the spec here though.  SCIM _requires_ a 501 because the SCIM authors are ... odd

Reply:
But maybe lets ignore their advice